### ISSUE TITLE
[Blazor] Lifecyle hooks not being called while prerendering

### ISSUE THREAD

**Issue by Gaulomatic** [aspnet/AspNetCore#9804]
### Describe the bug
Lifecycle hooks, like `OnInit`, are not being invoked when a component gets pre-rendered server-side. First invocation takes place after SignalR is connected and when the component in re-rendered.

### To Reproduce
1. Get the latest preview5 sdk from core-sdk
2. dotnet new blazorserverside
3. Override `OnInit` in `Counter` component and log something to the console
3. dotnet run
4. Open site in the browser, view source code only so that no SignalR connection is being established

### Expected behavior
Log output should appear in the console window. Lifecycle hooks should be invoked when a component is pre-rendering on the server - which was the case in preview 4. 

### Additional context

.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview5-011562
 Commit:    5346fb8ae6

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.14
 OS Platform: Darwin
 RID:         osx.10.14-x64
 Base Path:   /usr/local/share/dotnet/sdk/3.0.100-preview5-011562/

Host (useful for support):
  Version: 3.0.0-preview5-27624-11
  Commit:  7b5103a1f6

**Comment by danroth27** [comment]
@Gaulomatic You won't see the console output in the browser during prerender, because the code is running on the server. If you need to do anything that requires JS interop then you need to put that logic in the `OnAfterRenderAsync` method.

Since this sounds like it's by design, I'm going to go ahead and close. If we've misunderstood the issue please let us know!

**Comment by Gaulomatic** [comment]
I was referring to `System.Console.WriteLine()`, so on the server. This was working in preview4, but it does not in preview5 and preview6.

**Comment by danroth27** [comment]
@Gaulomatic I'm not able to reproduce this. I updated `Counter` like this:

```
@page "/counter"
@inject IComponentContext Context

<h1>Counter</h1>

<p>Current count: @currentCount</p>

<button class="btn btn-primary" onclick="@IncrementCount">Click me</button>

@functions {
    int currentCount = 0;

    void IncrementCount()
    {
        currentCount++;
    }

    protected override void OnInit()
    {
        Console.WriteLine("Counter.OnInit " + (Context.IsConnected ? "connected" : "prerendered"));
    }
}
```

If I then run the app and browse to view-source:https://localhost:5001/counter I get this in the console logs:

```
info: Microsoft.AspNetCore.Hosting.Diagnostics[1]
      Request starting HTTP/2 GET https://localhost:5001/counter
info: Microsoft.AspNetCore.Routing.EndpointMiddleware[0]
      Executing endpoint '/_Host'
info: Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker[3]
      Route matched with {page = "/_Host"}. Executing page /_Host
info: Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker[103]
      Executing an implicit handler method - ModelState is Valid
info: Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker[104]
      Executed an implicit handler method, returned result Microsoft.AspNetCore.Mvc.RazorPages.PageResult.
Counter.OnInit prerendered
info: Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure.PageActionInvoker[4]
      Executed page /_Host in 57.2432ms
info: Microsoft.AspNetCore.Routing.EndpointMiddleware[1]
      Executed endpoint '/_Host'
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
      Request finished in 60.8646ms 200 text/html; charset=utf-8
```

**Comment by Gaulomatic** [comment]
Weired, I can longer reproduce this any more. Yesterday I had to uninstall a bunch of other preview SDKs (5s and 6s) because I was no longer able to start anything other than preview6 stuff and now this weired behavoir is gone. I uninstalled essentially every 3.0 SDK except for preview4 and preview5-011562 and reverted the `dotnet` executable to preview5-011562. This must have been some installation mess up on my side. I am sorry that I wasted your time, next time I will check things on a second machine before I open a ticket.
